pkg/ui/app/components/graphs.tsx

```diff
@@ -77,59 +99,270 @@ export class Axis extends React.Component<AxisProps, {}> {
+  // the maximum value representing the top of the axis.
+  max: number = 1;
+  // numbers at which an intermediate tick should be displayed on the axis.
+  ticks: number[] = [0, 1];
+  // label returns the label for the axis.
+  label: string = "";
+  // tickFormat returns a function used to format the tick values for display.
+  tickFormat: (n: number) => any = _.identity;
+  // guideFormat returns a function used to format the axis values in the
+  // chart's interactive guideline.
+  guideFormat: (n: number) => any = _.identity;
 
-  // ticks computes tick values for a graph given the current max/min and 
-  // tickCount.
-  ticks(transform: (n: number) => any = _.identity): number[] {
-    let increment = (this.max - this.min) / (this.tickCount + 1);
-    let tix: number[] = [];
-    if (this.maxMinTicks) {
-      tix.push(this.min);
+  // constructs a new AxisRange with the given minimum and maximum value, with
+  // ticks placed at intervals of the given increment in between the min and
+  // max. Ticks are always "aligned" to values that are even multiples of
+  // increment. Min and max are also aligned by default - the aligned min will
+  // be <= the provided min, while the aligned max will be >= the provided max.
+  constructor(min: number, max: number, increment: number, alignMinMax: boolean = true) {
+    if (alignMinMax) {
+      this.min = min - min % increment;
+      this.max = max - max % increment + increment;
+    } else {
+      this.min = min;
+      this.max = max;
     }
-    for (let i = 0; i < this.tickCount; i++) {
-      tix.push(this.min + increment * (i + 1));
+
+    this.ticks = [];
+    for (let nextTick = min - min % increment + increment;
+         nextTick < this.max;
+         nextTick += increment) {
+      this.ticks.push(nextTick);
     }
-    if (this.maxMinTicks) {
-      tix.push(this.max);
+  }
+
+  domain() {
+    return [this.min, this.max];
+  }
+}
+
+const countIncrementTable = [0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.8, 0.9, 1.0];
+
+// computeNormalizedIncrement computes a human-friendly increment between tick
+// values on an axis with a range of the given size. The provided size is taken
+// to be the minimum range needed to display all values present on the axis.
+// The increment is computed by dividing this minimum range into the correct
+// number of segments for the supplied tick count, and then increasing this
+// increment to the nearest human-friendly increment.
+//
+// "Human-friendly" increments are taken from the supplied countIncrementTable,
+// which should include decimal values between 0 and 1.
+function computeNormalizedIncrement(
+  range: number, tickCount: number, incrementTbl: number[] = countIncrementTable
+) {
+  if (range === 0) {
+    throw new Error("cannot compute tick increment with zero range");
+  }
+
+  let rawIncrement = range / (tickCount + 1);
+  // Compute X such that 0 <= rawIncrement/10^x <= 1
+  let x = 0;
+  while (rawIncrement > 1) {
```

Comment:
I think this can be done in one step with logs as well if you want. But I suppose it doesn't read as clearly.

```x = Math.ceil(Math.log(rawIncrement)/Math.log(10))```